utilitiesApp/tests/calibration_range_tests.cc

```diff
@@ -5,12 +5,267 @@
 #include <vector>
 #include <stdio.h>
 #include <epicsString.h>
+#include <menuFtype.h>
+#include <errlog.h>
 
 #include "gtest/gtest.h"
+#include "find_calibration_range_impl.h"
 #include "find_calibration_range_utils.h"
 
 namespace {
 
+    double valaArray[1] = { 10000.0 };
+    double valbArray[1] = { 0.0 };
+	
+	static void setupASubRecord(aSubRecord* prec)
+    {
+        strcpy(prec->name, "name");
+        prec->fta = menuFtypeSTRING;
+        prec->ftb = menuFtypeSTRING;
+        prec->ftc = menuFtypeSTRING;
+        prec->a = ".";
+        prec->b = ".";
+        prec->c = "calibration_range.txt";
+        prec->ftva = menuFtypeDOUBLE;
+        prec->ftvb = menuFtypeDOUBLE;
+        prec->vala = valaArray;
+        prec->valb = valbArray;
+    }
+
+    static void createCalibrationFile(std::string data = "0.0\n10000.0") {
+        std::FILE* pFile;
+        pFile = fopen("calibration_range.txt", "w");
+
+        if (pFile != NULL) {
+            std::fputs(data.c_str(), pFile);
+            std::fclose(pFile);
+        }
+    }
+
+    TEST(FindCalibrationRangeTest, reads_in_calibration_range) {
+        // Given
+        aSubRecord prec;
+        memset(&prec, 0, sizeof(prec));
+        setupASubRecord(&prec);
+
+        createCalibrationFile();
+        
+        // When:
+        long status = find_calibration_range_impl(&prec);
+
+        // Then:
+        ASSERT_EQ(status, 0);
+        ASSERT_EQ((static_cast<double*>(prec.vala))[0], 10000);
+        ASSERT_EQ((static_cast<double*>(prec.valb))[0], 0);
+
+        // Clean up
+        std::remove(std::string{"calibration_range.txt"}.c_str());
```

Comment:
No need to create a `std::string` and cast it to a c-string. I would advise just defining a macro or a `const` value of your file name and using that instead of having to pass a literal each time.